### ISSUE TITLE
Focus shifts and cueing hand may reappear when switching between keyboard and pointer

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/mean-share-and-balance#343]
**Test device**
MacBook Air M1 chip

**Operating System**
14.5

**Browser**
Safari 17.5

**Problem description**
For https://github.com/phetsims/qa/issues/1134 and related to https://github.com/phetsims/mean-share-and-balance/issues/329:
On the Balance Point Screen when switching from keyboard to trackpad, the focus highlight on the soccer balls sometimes shifts.
Also, on both the Balance Point and Distribute Screens, the cueing had can reappear if no balls/bars had been moved before switching from keyboard to trackpad. 

**Steps to reproduce**
For Focus Changing:
1. On the Balance Point Screen, press and hold the increment button with the point until it reads 7
2. While the balls are still being kicked, tab to one of the soccer balls
3. Once all the balls are kicked, grab and move that ball
4. While that ball still has focus, move the pointer over another ball --keyboard focus highlight will move

For Cueing Hand reappearing:
1. On the Balance Point Screen, press and hold the increment button with the point until it reads 7
2. Wait until all balls are kicked
3. Tab to and Grab the last ball  kicked but don't move it
4. While that ball still has focus, move the pointer over another ball --the cueing hand will reappear
OR
1. On the Distribute Screen, tab to and grab the bar with the cueing arrows
2. Move pointer over a bar

**Visuals**

https://github.com/user-attachments/assets/20727016-27b7-4970-87d5-bf730a2e81bd

https://github.com/user-attachments/assets/a539e084-4e31-4d2a-ba35-cc2fb3da714a

https://github.com/user-attachments/assets/c1db0787-52f9-4a7d-94b3-a5afe809963f

**Comment by marlitas** [comment]
The cueing hand reappearing when mouse is hovered over the ball group is expected behavior. This behavior was implemented and can also be seen in CAV for both the soccer balls and card sorting. That is a won't fix.

I will focus on debugging why the focus shifts during mouse hovering.

**Comment by marlitas** [comment]
Hit an assertion in my first round of reproducing... So there is additional work to look at here.
<img width="1753" alt="image" src="https://github.com/user-attachments/assets/e1cd6130-289f-487b-bd02-46f50214d5f0">

**Comment by marlitas** [comment]
The moving focus bug has been fixed in the above commits. Ready for cherry-pick.

**Comment by zepumph** [comment]
I believe that cherry picking this issue should be blocked by the review over in https://github.com/phetsims/scenery-phet/issues/841, since https://github.com/phetsims/soccer-common/commit/a1bc6b891f2c3d75cfbc42069c06c6b49c7f6417 from above is actually onto changes from that other issue.

**Comment by KatieWoe** [comment]
The focus shift and assertion issues seem fixed in rc.3
